CMakeLists.txt

```diff
@@ -689,14 +701,35 @@ if(UA_ENABLE_TPM2_SECURITY)
 endif()
 
 if(UA_ENABLE_WEBSOCKET_SERVER)
-    # The recommended way is to install libwebsockets via the OS package manager. If
-    # that is not possible, manually compile libwebsockets and set the cmake variables
-    # defined in /tools/cmake/Findlibwebsockets.cmake
+   if(NOT UA_ENABLE_LWS)
+       option(UA_ENABLE_LWS ON)
+   endif()
+endif()
+
+#Libwebsockets
+if(UA_ENABLE_LWS)
     find_package(libwebsockets REQUIRED)
-    list(APPEND open62541_LIBRARIES ${LIBWEBSOCKETS_LIBRARIES})
+    if(LIBWEBSOCKETS_FOUND)
+        if(${LIBWEBSOCKETS_VERSION} VERSION_LESS "4.3")
+            message(STATUS "The version of libwebsockets is too low (required v4.3). Instead, the library is used in the dependencies.")
+            set(LIBWEBSOCKETS_FOUND 0)
+        endif()
+    endif()
+    if(NOT LIBWEBSOCKETS_FOUND)
```

Comment:
Throw an exception when not found.
The `find_package` thing allows users to manually define the folders.
That should be the only way of doing things.
Like for OpenSSL/mbedTLS.

Reply:
Do not modify settings.
Abort with a warning when you cannot build.